aya/src/programs/links.rs

```diff
@@ -381,6 +381,24 @@ macro_rules! define_link_wrapper {
                 w.0.take().unwrap()
             }
         }
+
+        impl $program {
+            /// Detaches the program.
+            ///
+            /// See [`Self::attach`].
+            pub fn detach(&mut self, link_id: $wrapper_id) -> Result<(), ProgramError> {
+                self.data.links.remove(link_id)
+            }
+
+            /// Takes ownership of the link referenced by the provided `link_id`.
+            ///
+            /// The caller takes the responsibility of managing the lifetime of the
+            /// link. When the returned [`$wrapper_id`] is dropped, the link is
```

Comment:
```suggestion
            /// link. When the returned [`$wrapper`] is dropped, the link is
```

We return the `Link`, not the `LinkId`.

Reply:
Thanks! Done.